specs/graph-protocol-hybrid-network/read-interface/README.md

```diff
@@ -1,357 +1,273 @@
-  id: ID!
-  address: String!
-  name: FullName
-}
+The entire query follows this grammar:
 
-type FullName {
-  first: String!
-  last: String!
-}
+```
+    QUERY ::= select FIELD [, FIELD]*
+                from TABLE_REF
+                     [[inner|left outer] join] TABLE_REF on JOINCOND
+              [where PREDICATE]
+              [order by ATTR [asc|desc]]
+              [limit INT]
+            | QUERY union all QUERY
 ```
 
-Then, the following would be valid index names for that dataset:
+with the following nonterminals:
 
-| Index Name | Description|
-| ---------- | ---------- |
-| `{ db: "entitydb", indexType: "dictionary" }`       | A basic key-value index supporting constant-time lookup of all entities in dataset. |
-| `{ db: "entitydb", "indexType: "dictionary", partition: "User" }`  | A basic key-value index supporting constant-time lookup of `User` entities. |
-| `{ db: "entitydb", indexType: "searchTree", options: { sortBy: ["id"] } }` | A sorted key-value index supporting iteration through all entities, sorted by ID. |
-| `{ db: "entitydb", indexType: "searchTree", partition: "EthereumAccount", options: { sortBy: ["address"] }}` |  A sorted key-value index supporting iteration through all entities implementing the `EthereumAccount` interface, sorted by the `address` field. |
-| `{ db: "entitydb", indexType: "searchTree", partition: "User",  options: { sortBy: ["name.first", "name.last"] } }` | A sorted key-value index supporting iteration through all `User` entities, first sorted by the nested field `name.first`, then by the nested field `name.last` (i.e., a compound index). |
+```
+    TABLE_REF ::= NAME [as NAME]
+                | (QUERY) as NAME
+    FIELD     ::= ATTR [as NAME]
+                | map(NAME) as NAME
+                | collect(NAME) as NAME over NAME [, NAME ]*
+    ATTR      ::= NAME . NAME
+    JOINCOND  ::= ATTR = ATTR [and ATTR = ATTR]*
+    PREDICATE ::= ATTR OP VALUE
+                | VALUE OP ATTR
+                | PREDICATE and PREDICATE
+                | PREDICATE or PREDICATE
+                | not PREDICATE
+                | (PREDICATE)
+    OP        ::= = | != | < | <= | > | >=
+                | in
+                | contains | starts with | ends with
+    VALUE     ::= 'STRING' | INT | ( VALUE [, VALUE ]* )
+                | literals for whatever other types we'll support
+    NAME      ::= /[A-Za-z][A-Za-z_]*/
+    INT       ::= /[0-9]+/
 
-### Index Abstract Data Structures
+```
 
-All concrete index types implement an indexing abstract data structure, which specify the interface, semantics, and gas costs for read operations against that index.
+### AST representation
 
-The concrete types (i.e., `K` and `V` shown below), as well as the implicit comparator function to determine sort order, are specified by each concrete index type.
+Queries are transmitted as abstract syntax trees, not as text. The precise representation of the abstract syntax tree in JSON still needs to be defined.
```

Comment:
Same comment as above. Let's link to an issue if we are going to list this as a TODO in the spec.

Reply:
Tracked here: https://github.com/graphprotocol/graph-node/issues/1097